coordinator/src/main/java/io/narayana/lra/coordinator/management/BrowserCommand.java

```diff
@@ -0,0 +1,450 @@
+            return outputStream.toString();
+        }
+    }
+
+    CommandName name;
+    boolean verbose;
+
+    private BrowserCommand(CommandName name) {
+        this.name = name;
+    }
+
+    abstract void execute(PrintStream printStream, List<String> args) throws Exception;
+
+    protected void help(PrintStream printStream) {
+        if (name.cmdHelp != null)
+            printStream.printf("%s - %s%n", name.name().toLowerCase(), name.cmdHelp);
+    }
+
+    boolean cancel() {return true;}
+
+    private static void setupStore(String storeDir) throws Exception {
+        recoveryPropertyManager.getRecoveryEnvironmentBean().setRecoveryBackoffPeriod(1);
+        Implementations.install();
+        recoveryManager = new RecoveryManagerImple(false);
+        recoveryManager.addModule(new LRARecoveryModule());
+
+        setupStore(storeDir, false);
+
+        osb = new ObjStoreBrowser();
+        for(String[] typeAndBean: LRA_OSB_TYPES) {
+            osb.addType(typeAndBean[0], typeAndBean[1], typeAndBean[2]);
+        }
+        osb.start();
+    }
+
+    private static void setupStore(String storeDir, boolean hqstore) throws Exception {
+        String storePath = new File(storeDir).getCanonicalPath();
+
+        BeanPopulator.getDefaultInstance(ObjectStoreEnvironmentBean.class).setObjectStoreDir(storePath);
+        BeanPopulator.getNamedInstance(ObjectStoreEnvironmentBean.class, "communicationStore").setObjectStoreDir(storePath);
+        BeanPopulator.getDefaultInstance(CoreEnvironmentBean.class).setNodeIdentifier("no-recovery");
+
+        currentStoreDir = storeDir;
+    }
+
+    private static void restartStore(String storeDir) throws Exception {
+        StoreManager.shutdown();
+
+        if (osb != null)
+            osb.stop();
+
+        try {
+            RecoveryManager.manager().terminate(false);
+        } catch (Throwable ignore) {
+        }
+
+        setupStore(storeDir);
+    }
+
+    private static final BrowserCommand[] commands = {
+
+            new BrowserCommand(CommandName.HELP) {
+                @Override
+                void execute(PrintStream printStream, List<String> args) {
+                    for (BrowserCommand command : commands)
+                        command.help(printStream);
+                }
+            },
+
+            new BrowserCommand(CommandName.START) {
+                boolean finished;
+
+                @Override
+                void execute(PrintStream printStream, List<String> args) throws Exception {
+                    setupStore(currentStoreDir);
+
+                    Scanner scanner = new Scanner(cmdSource);
+
+                    getCommand(CommandName.PROBE).execute(printStream, null);
+                    getCommand(CommandName.TYPES).execute(printStream, null);
+
+                    while (!finished)
+                        processCommand(printStream, scanner);
+
+                    scanner.close();
+
+//                    RecoveryManager.manager().terminate();
```

Comment:
Maybe this can be removed? Or is this part work in progress?

Reply:
I can't get the store to change once it's been initialised so it is work in progress but is likely to change if I can figure out how to reinitialise the store.